Code/GraphMol/SmilesParse/CXSmilesOps.cpp

```diff
@@ -414,6 +414,70 @@ std::string quote_string(const std::string &txt) {
   // FIX
   return txt;
 }
+
+std::string get_enhanced_stereo_block(const ROMol &mol,
+                            const std::vector<unsigned int> &atomOrder){
+  std::string res = "";
+  // we need a map from original atom idx to output idx:
+  std::vector<unsigned int> revOrder(mol.getNumAtoms());
+  for(unsigned i=0;i<mol.getNumAtoms();++i){
+    revOrder[atomOrder[i]] = i; 
+  }
+  std::vector<unsigned int> absAts;
+  std::vector<std::vector<unsigned int>> orGps;
+  std::vector<std::vector<unsigned int>> andGps;
+
+  // we want this to be canonical (future proofing)
+  for(const auto &sg : mol.getStereoGroups()) {
+    std::vector<unsigned int> aids;
+    aids.reserve(sg.getAtoms().size());
+    for(const auto at : sg.getAtoms()){
+      aids.push_back(revOrder[at->getIdx()]);
+    }  
+    std::sort(aids.begin(),aids.end());
+    switch(sg.getGroupType()){
+    case StereoGroupType::STEREO_ABSOLUTE:
+      absAts.push_back(aids[0]);
+      break;
+    case StereoGroupType::STEREO_OR:
+      orGps.push_back(aids);
+      break;
+    case StereoGroupType::STEREO_AND:
+      andGps.push_back(aids);
+      break;  
+    }
+  }
+  if(!absAts.empty()){
+    res += "a:";
+    std::sort(absAts.begin(),absAts.end());
```

Comment:
This shouldn't be needed, as there can be only one absolute group, and its atoms have already been sorted.

Reply:
I don't think we enforce that anywhere (and am not sure that we should), so it seems like allowing multiple potential ABS groups to be fused on writing would be helpful.